qiskit/pulse/transforms.py

```diff
@@ -236,3 +261,127 @@ def compress_pulses(schedules: List[Schedule]) -> List[Schedule]:
+        Joined schedule
+    """
+    this_channels = set(this.channels)
+    other_channels = set(other.channels)
+    shared_channels = list(this_channels & other_channels)
+    ch_slacks = [this.stop_time - this.ch_stop_time(channel) + other.ch_start_time(channel)
+                 for channel in shared_channels]
+
+    if ch_slacks:
+        slack_chan = shared_channels[np.argmin(ch_slacks)]
+        shared_insert_time = this.ch_stop_time(slack_chan) - other.ch_start_time(slack_chan)
+    else:
+        shared_insert_time = 0
+
+    other_only_insert_time = other.ch_start_time(*(other_channels - this_channels))
+    insert_time = max(shared_insert_time, other_only_insert_time)
+    return this.insert(insert_time, other, mutate=True)
+
+
+def align_left(schedule: Schedule) -> Schedule:
+    """Align a list of pulse instructions on the left.
+
+    Args:
+        schedule: Input schedule of which top-level ``child`` nodes will be
+            reschedulued.
+
+    Returns:
+        New schedule with input `schedule`` child schedules and instructions
+        left aligned.
+    """
+    aligned = Schedule()
+    for _, child in schedule.children:
+        _push_append(aligned, child)
+
+    return aligned
+
+
+def align_right(schedule: Schedule) -> Schedule:
+    """Align a list of pulse instructions on the right.
+
+    Args:
+        schedule: Input schedule of which top-level ``child`` nodes will be
+            reschedulued.
+
+    Returns:
+        New schedule with input `schedule`` child schedules and instructions
+        right aligned.
+    """
+    left_aligned = align_left(schedule)
+    total_duration = left_aligned.duration
+
+    latest_available_times = collections.defaultdict(lambda: total_duration)
+    right_aligned = Schedule()
+    for _, child in reversed(left_aligned.children):
+        child_channels = child.channels
+        # drop node without channels as it is empty
+        if child_channels:
+            latest_available_duration = min(latest_available_times[channel] for
+                                            channel in child_channels)
+            insert_time = latest_available_duration - child.duration
+            right_aligned.insert(insert_time, child, mutate=True)
+            latest_available_times.update(
+                dict.fromkeys(child_channels, insert_time))
+
+    return right_aligned
+
+
+def align_sequential(schedule: Schedule) -> Schedule:
+    """Schedule all top-level nodes in parallel.
+
+    Args:
+        schedule: Input schedule of which top-level ``child`` nodes will be
+            reschedulued.
+
+    Returns:
+        New schedule with input `schedule`` child schedules and instructions
+        applied sequentially across channels
+    """
+    aligned = Schedule()
+    for _, child in schedule.children:
+        aligned.insert(aligned.duration, child, mutate=True)
+    return aligned
+
+
+def group(schedule: Schedule) -> Schedule:
+    """Group all instructions in a node.
```

Comment:
What's a node, here? A block?

Reply:
I suppose barriers will be inserted here.

Reply:
Removed for now. See Lauren's other comment on this for my response. This should be added back at a later date.